code/sonalyze/daemon/perform.go

```diff
@@ -0,0 +1,387 @@
+package daemon
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log/syslog"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	"go-utils/auth"
+	"go-utils/httpsrv"
+	"go-utils/process"
+)
+
+func (dc *DaemonCommand) RunDaemon(_ io.Reader, _, stderr io.Writer) /* never returns */ {
+	if err := statusStart("jobanalyzer/sonalyzed", stderr); err != nil {
+		fmt.Fprintf(stderr, "FATAL ERROR: Failing to open logger: %s", err.Error())
+		os.Exit(1)
+	}
+
+	// Note "daemon" is not a command here
+	http.HandleFunc("/add", httpAddHandler(dc))
+	http.HandleFunc("/jobs", httpGetHandler(dc, "jobs"))
+	http.HandleFunc("/load", httpGetHandler(dc, "load"))
+	http.HandleFunc("/uptime", httpGetHandler(dc, "uptime"))
+	http.HandleFunc("/profile", httpGetHandler(dc, "profile"))
+	http.HandleFunc("/parse", httpGetHandler(dc, "parse"))
+	http.HandleFunc("/metadata", httpGetHandler(dc, "metadata"))
```

Comment:
It's extremely questionable whether /metadata should be a command here, as it will reveal absolute paths for files on the server.  We should probably comment this out.  There might be a similar change to be made in the remotable handling, then.

Reply:
Filed a bug.